Source/Core/VideoCommon/BPFunctions.h

```diff
@@ -7,16 +7,132 @@
+        scissor_off{.hex = other.scissor_off.hex}, viewport_left{other.viewport_left},
+        viewport_right{other.viewport_right}, viewport_top{other.viewport_top},
+        viewport_bottom{other.viewport_bottom}, m_result{other.m_result}
+  {
+  }
+  ScissorResult& operator=(const ScissorResult& other)
+  {
+    if (this == &other)
+      return *this;
+    scissor_tl.hex = other.scissor_tl.hex;
+    scissor_br.hex = other.scissor_br.hex;
+    scissor_off.hex = other.scissor_off.hex;
+    viewport_left = other.viewport_left;
+    viewport_right = other.viewport_right;
+    viewport_top = other.viewport_top;
+    viewport_bottom = other.viewport_bottom;
+    m_result = other.m_result;
+    return *this;
+  }
+  ScissorResult(ScissorResult&& other)
+      : scissor_tl{.hex = other.scissor_tl.hex}, scissor_br{.hex = other.scissor_br.hex},
+        scissor_off{.hex = other.scissor_off.hex}, viewport_left{other.viewport_left},
+        viewport_right{other.viewport_right}, viewport_top{other.viewport_top},
+        viewport_bottom{other.viewport_bottom}, m_result{std::move(other.m_result)}
+  {
+  }
+  ScissorResult& operator=(ScissorResult&& other)
+  {
+    if (this == &other)
+      return *this;
+    scissor_tl.hex = other.scissor_tl.hex;
+    scissor_br.hex = other.scissor_br.hex;
+    scissor_off.hex = other.scissor_off.hex;
+    viewport_left = other.viewport_left;
+    viewport_right = other.viewport_right;
+    viewport_top = other.viewport_top;
+    viewport_bottom = other.viewport_bottom;
+    m_result = std::move(other.m_result);
+    return *this;
+  }
+
+  // Input values, for use in statistics
+  ScissorPos scissor_tl;
+  ScissorPos scissor_br;
+  ScissorOffset scissor_off;
+  float viewport_left;
+  float viewport_right;
+  float viewport_top;
+  float viewport_bottom;
+
+  // Actual result
+  std::vector<ScissorRect> m_result;
+
+  ScissorRect Best() const;
+
+  bool ScissorMatches(const ScissorResult& other) const
+  {
+    return scissor_tl.hex == other.scissor_tl.hex && scissor_br.hex == other.scissor_br.hex &&
+           scissor_off.hex == other.scissor_off.hex;
+  }
+  bool ViewportMatches(const ScissorResult& other) const
+  {
+    return viewport_left == other.viewport_left && viewport_right == other.viewport_right &&
+           viewport_top == other.viewport_top && viewport_bottom == other.viewport_bottom;
+  }
+  bool Matches(const ScissorResult& other, bool compare_scissor, bool compare_viewport) const
+  {
+    if (compare_scissor && !ScissorMatches(other))
+      return false;
+    if (compare_viewport && !ViewportMatches(other))
+      return false;
+    return true;
+  }
+
+private:
+  ScissorResult(const BPMemory& bpmem, std::pair<float, float> viewport_x,
+                std::pair<float, float> viewport_y);
+
+  int GetViewportArea(const ScissorRect& rect) const;
+  bool IsWorse(const ScissorRect& lhs, const ScissorRect& rhs) const;
```

Comment:
Should this be a static member function?

Reply:
It can't be static, because it needs to call `GetViewportArea` which is non-static.